src/js/controller/instream-adapter.js

```diff
@@ -192,10 +195,33 @@ const InstreamAdapter = function(_controller, _model, _view, _mediaPool) {
         }
     }
 
+    function _triggerAdPlayPause(event) {
+        const { newstate } = event;
+        const adEvent = Object.assign({
+            oldstate: _adProgram.model.get('state')
+        }, _data, event);
+
+        if (adEvent.oldstate === newstate) {
```

Comment:
I would prefer that this check be performed before creating `adEvent`.
```js
const { newstate } = event;
const oldstate = _adProgram.model.get('state');
if (oldstate === newstate) {
  return;
...
const adEvent = Object.assign({ oldstate }, _data, event);
...
```

Reply:
The `oldstate` might come be set on [`event`](https://github.com/jwplayer/jwplayer/pull/3169/files#diff-4b5d9c794240aa20cb620353eb027d56R114) though.